src/main/scala/org/scalameter/Executor.scala

```diff
@@ -30,21 +32,22 @@ object Executor {
   import Key._
 
   type Warmer = org.scalameter.Warmer
-  type Measurer = org.scalameter.Measurer
+  type Measurer[T] = org.scalameter.Measurer[T]
 
   val Warmer = org.scalameter.Warmer
   val Measurer = org.scalameter.Measurer
 
-  object None extends Executor {
-    def measurer: Measurer = ???
+  object None extends Executor[Nothing] {
+    def measurer: Measurer[Nothing] = ???
 
     def warmer: Warmer = ???
 
-    def runSetup[T](setup: Setup[T]): CurveData = ???
+    def runSetup[T](setup: Setup[T]): CurveData[Nothing] = ???
   }
 
-  trait Factory[E <: Executor] {
-    def apply(warmer: Warmer, aggregator: Aggregator, m: Measurer): E
+  trait Factory[E[_] <: Executor[_]] {
+    def apply[T: Numeric: Pickler](warmer: Warmer,
```

Comment:
Since `aggregator` is coming here externally, it seems to me that we don't need a `Numeric` typeclass here, or the `Pickler` - they should be provided at the instantiation site when the `aggregator` and `Measurer` are created. Or am I missing something?